tools/crm_mon.c

```diff
@@ -2206,23 +2207,30 @@ mon_refresh_display(gpointer user_data)
     switch (output_format) {
         case mon_output_html:
         case mon_output_cgi:
-            if (print_html_status(mon_data_set, crm_errno2exit(history_rc),
-                                  stonith_history, fence_history,
-                                  get_resource_display_options(),
-                                  show, options.neg_location_prefix,
-                                  unames, resources) != 0) {
+            if (out->message(out, "cluster-status", mon_data_set,
+                             crm_errno2exit(history_rc), stonith_history,
+                             fence_history, get_resource_display_options(),
+                             show, options.neg_location_prefix, unames,
+                             resources) != pcmk_rc_ok) {
                 g_set_error(&error, PCMK__EXITC_ERROR, CRM_EX_CANTCREAT, "Critical: Unable to output html file");
                 clean_up(CRM_EX_CANTCREAT);
                 return 0;
             }
             break;
 
+        case mon_output_console:
         case mon_output_legacy_xml:
+        case mon_output_plain:
         case mon_output_xml:
-            print_xml_status(mon_data_set, crm_errno2exit(history_rc),
-                             stonith_history, fence_history,
-                             get_resource_display_options(), show,
-                             options.neg_location_prefix, unames, resources);
+#if CURSES_ENABLED
+            blank_screen();
+#endif
```

Comment:
You don't need to guard blank_screen() with CURSES_ENABLED since it has that internally

However we don't want to call blank_screen() and refresh() for anything but mon_output_console. A better way to do this would be to implement a console version of cluster-status. Then this simplifies dramatically

Reply:
Fixed.  Yeah that is much better.